lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -784,6 +784,9 @@ static Optional<RequirementMatch> findMissingGenericRequirementForSolutionFix(
   default:
     return Optional<RequirementMatch>();
   }
+  
+  if (missingType->hasTypeVariable())
```

Comment:
Would it be possible to actually provide a solution here and use `solution.simplifyType` on `missingType`? That should make sure that there are no type variables.

Reply:
Solution associated with the fix considered that is.

Reply:
Will a damaged solution always substitute out every type variable?

Reply:
Yes, if there was a fixed solution produced it would not leave any type variables after substitution.

Reply:
Fantastic. Then it seems like we just have to substitute in each one of these little closures that act as shared return points and assert that we're not leaking type variables.

Reply:
Yeah, that would make sense!